src/System.ServiceModel.Syndication/tests/BasicScenarioTests.cs

```diff
@@ -368,43 +358,130 @@ public static void AtomEntryPositiveTest_write()
         }
 
         [Fact]
-        [ActiveIssue(25058)]
-        public static void AtomFeedPositiveTest_WsFeedIcon()
+        public static void AtomFeedPositiveTest()
         {
-            string file = "ws-feed-icon.xml";
-            AtomFeedPositiveTest(file);
+            string dataFile = @"atom_feeds.dat";
+            List<string> fileList = GetTestFilesForFeedTest(dataFile);
+            List<AllowableDifference> allowableDifferences = GetAtomFeedPositiveTestAllowableDifferences();
+
+            foreach (string file in fileList)
+            {
+                FeedPositiveTest(file, FeedFormatterType.Atom10FeedFormatter, allowableDifferences);
+            }
         }
 
         [Fact]
-        public static void AtomFeedPositiveTest()
+        public static void RssEntryPositiveTest()
         {
-            string dataFile = @"atom_feeds.dat";
+            string file = @"RssEntry.xml";
+            EntryPositiveTest(file, SyndicationItemFormatterType.Rss20ItemFormatter);
+        }
+
+        [Fact]
+        public static void RssFeedPositiveTest()
+        {
+            string dataFile = @"rss_feeds.dat";
             List<string> fileList = GetTestFilesForFeedTest(dataFile);
+            List<AllowableDifference> allowableDifferences = GetRssFeedPositiveTestAllowableDifferences();
 
             foreach (string file in fileList)
             {
-                AtomFeedPositiveTest(file);
+                FeedPositiveTest(file, FeedFormatterType.Rss20FeedFormatter, allowableDifferences);
             }
         }
 
-        private static void AtomFeedPositiveTest(string file)
+        private static void EntryPositiveTest(string file, SyndicationItemFormatterType syndicationItemFormatterType)
+        {
+
+            string serializeFilePath = Path.GetTempFileName();
+            try
+            {
+                SyndicationItem feedObjct = null;
+                using (FileStream fileStream = new FileStream(file, FileMode.Open, FileAccess.Read))
+                {
+                    XmlReader reader = XmlDictionaryReader.CreateTextReader(fileStream, XmlDictionaryReaderQuotas.Max);
+                    feedObjct = SyndicationItem.Load(reader);
+                    reader.Close();
+                }
+
+                using (FileStream fileStream = new FileStream(serializeFilePath, FileMode.OpenOrCreate))
+                {
+                    XmlWriter writer = XmlDictionaryWriter.CreateTextWriter(fileStream);
```

Comment:
Nit: we can use `using` so that we don't need to call Flush or Close directly.

Reply:
Done.